### PR TITLE
Added eigensolver miniapps

### LINKED ISSUE #521: Miniapp (standard) eigensolver

### PR DESCRIPTION
**Pull request by rasolca**
Closes #521 and #520.

Tests are not implemented yet. They can easily implemented with GEMM when we implement it.

### CODE REVIEW

**Review by msimberg — APPROVED** [review]
Can't comment on what it's actually doing for the miniapp, but looks good to me otherwise.

**Comment by rasolca** [comment]
bors try

**Review by albestro — APPROVED** [review]
LGTM

File: miniapp/miniapp_eigensolver.cpp [hunk 1]
```diff
@@ -0,0 +1,173 @@
...
+  Options(const pika::program_options::variables_map& vm)
+      : MiniappOptions(vm), m(vm["matrix-size"].as<SizeType>()), mb(vm["block-size"].as<SizeType>()),
+        uplo(dlaf::miniapp::parseUplo(vm["uplo"].as<std::string>())) {
+    DLAF_ASSERT(m > 0, m);
+    DLAF_ASSERT(mb > 0, mb);
+
+    if (do_check != dlaf::miniapp::CheckIterFreq::None) {
+      std::cerr << "Warning! At the moment result checking it is not implemented." << std::endl;
+      do_check = dlaf::miniapp::CheckIterFreq::None;
+    }
```
Comment (albestro) [inline miniapp/miniapp_eigensolver.cpp:61]: Just one comment/doubt.

For `--check-result` there is both the check in the `Options` c'tor and the `DLAF_UNIMPLEMENTED`. For "local" check there is just this unimplemented. Wouldn't be better to add a check in `Options` constructor too with the hint about how to setup the flags?

Something like this in 

```
DLAF_ASSERT((vm["grid-rows"].as<int>() * vm["grid-cols"].as<int>()) == 1,
              "Error! Distributed eigensolver is not avilable yet. "
              "Please rerun with both --grid-rows and --grid-cols set to 1");
```

**Review by albestro — APPROVED** [review]

**Comment by rasolca** [comment]
bors try

**Pull request merged**